core/embed/rust/src/ui/model_mercury/component/hold_to_confirm.rs

```diff
@@ -0,0 +1,331 @@
+use crate::{
+    time::Duration,
+    translations::TR,
+    ui::{
+        component::{Component, Event, EventCtx},
+        display::Color,
+        geometry::{Alignment2D, Offset, Rect},
+        lerp::Lerp,
+        shape,
+        shape::Renderer,
+    },
+};
+
+use super::{theme, Button, ButtonContent, ButtonMsg};
+
+#[cfg(feature = "haptic")]
+use crate::trezorhal::haptic::{self, HapticEffect};
+use crate::{
+    time::Stopwatch,
+    ui::{
+        component::Label,
+        constant::screen,
+        geometry::{Alignment, Point},
+        model_mercury::theme::{CONFIRM_ANIMATION_OFFSET, TITLE_HEIGHT},
+    },
+};
+use pareen;
+
+#[derive(Default, Clone)]
+struct HoldToConfirmAnim {
+    pub timer: Stopwatch,
+}
+
+impl HoldToConfirmAnim {
+    const DURATION_MS: u32 = 2200;
+
+    pub fn is_active(&self) -> bool {
+        self.timer
+            .is_running_within(Duration::from_millis(Self::DURATION_MS))
+    }
+
+    pub fn eval(&self) -> f32 {
+        self.timer.elapsed().to_millis() as f32 / 1000.0
+    }
+
+    pub fn get_parent_cover_opacity(&self, t: f32) -> u8 {
+        let parent_cover_opacity = pareen::constant(0.0).seq_ease_in_out(
+            0.0,
+            easer::functions::Cubic,
+            0.2,
+            pareen::constant(1.0),
+        );
+
+        u8::lerp(0, 255, parent_cover_opacity.eval(t))
+    }
+
+    pub fn get_header_opacity(&self, t: f32) -> u8 {
+        let header_opacity = pareen::constant(0.0).seq_ease_out(
+            0.1,
+            easer::functions::Cubic,
+            0.3,
+            pareen::constant(1.0),
+        );
+
+        u8::lerp(0, 255, header_opacity.eval(t))
+    }
+
+    pub fn get_header_opacity2(&self, t: f32) -> u8 {
+        let header_opacity2 = pareen::constant(1.0).seq_ease_in(
+            2.0,
+            easer::functions::Cubic,
+            0.2,
+            pareen::constant(0.0),
+        );
+
+        u8::lerp(0, 255, header_opacity2.eval(t))
+    }
+
+    pub fn get_circle_opacity(&self, t: f32) -> u8 {
+        let circle_opacity = pareen::constant(1.0).seq_ease_out(
+            0.1,
+            easer::functions::Cubic,
+            0.1,
+            pareen::constant(0.0),
+        );
+
+        u8::lerp(0, 255, circle_opacity.eval(t))
+    }
+
+    pub fn get_pad_color(&self, t: f32) -> Color {
+        let pad_color = pareen::constant(0.0).seq_ease_in_out(
+            0.1,
+            easer::functions::Cubic,
+            1.9,
+            pareen::constant(1.0),
+        );
+
+        Color::lerp(theme::GREY_EXTRA_DARK, theme::GREEN, pad_color.eval(t))
+    }
+
+    pub fn get_circle_max_height(&self, t: f32) -> i16 {
+        let circle_max_height = pareen::constant(0.0).seq_ease_in(
+            0.1,
+            easer::functions::Cubic,
+            1.5,
+            pareen::constant(1.0),
+        );
+
+        i16::lerp(266, 0, circle_max_height.eval(t))
```

Comment:
Can you justify `266`?

Reply:
see Scale - Circle Mask https://cloud.protopie.io/p/0690ddba77a6dfc833fca55b/r/8eed2702